### ISSUE TITLE
StreamableFile errors crash server

### ISSUE THREAD

**Issue by Jollarvia** [nestjs/nest#9938]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Here are two simple routes:
```
import { createReadStream } from 'fs';
@Controller()
export class AppController {
@Get("crash")
  async crash(@Res({ passthrough: true }) res, @Param() params): Promise<StreamableFile> {
    throw "crash";
  }

  @Get("crash-server")
  async crashServer(@Res({ passthrough: true }) res, @Param() params): Promise<StreamableFile> {
    return new StreamableFile(createReadStream('pathToFileThatDoesNotExist'));
  }
}
```

The first crashes in the built in exception layer and returns a 500 error code to the user which I expect.
The second, from what I read about StreamableFile, the delivery of the file must be handled down stream from the error handling layer and apparently has no error handling of its own because this crashes the server. No error is returned to the user and no subsequent requests can be served.

For useability, it should not be incumbent on the developer to handle an error in the provided stream just to maintain the server. Given that NestJS handles other errors within the request it should also be able to handle this.

### Minimum reproduction code

https://codesandbox.io/s/pensive-shape-htlmr7?file=/src/app.controller.ts

### Steps to reproduce

1. Create the controller
2. Visit the endpoint "crash-server"
3. The server crashes.

### Expected behavior

The error in the stream is minimally handled and a 500 error is returned to the user. The server remains viable.

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

8.4.6

### Packages versions

"@nestjs/common": "^8.4.6",
"@nestjs/core": "^8.4.6",

### Node.js version

16.15.0

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
I guess this will be fixed by PR #9819

![image](https://user-images.githubusercontent.com/13461315/179424656-ddbf8e5f-bdfb-41af-825d-3f55ae766516.png)

**Comment by Jollarvia** [comment]
> I believe this will be fixed by PR #9819
> 
> ![image](https://user-images.githubusercontent.com/13461315/179424705-18167038-02aa-4b2a-af0a-5108cbff0852.png)
> 
> (at least for express platform)

This does seem promising.

**Comment by kamilmysliwiec** [comment]
Let's track this here https://github.com/nestjs/nest/pull/9819
